### ISSUE TITLE
Capacity is throwing TypeError: '>=' not supported between instances of 'float' and 'NoneType'

### ISSUE THREAD

**Issue by Cerlancism** [mobeets/bpcs#3]
Encode and decode works fine. However when I tried to get the capacity from images, including the given example, it throws
`TypeError: '>=' not supported between instances of 'float' and 'NoneType'`

`$ python -m bpcs.bpcs capacity -i examples/vessel.png -a 0.45`

Output:
```
Slicing...
Graying...
Loaded image as array with shape (300, 300, 3, 8)
Creating histograms of image complexity...
Found 34656 grids
D:\Documents\GitHub\bpcs\bpcs\bpcs_steg_capacity.py:15: FutureWarning: Using a non-tuple sequence for multidimensional indexing is deprecated; use `arr[tuple(seq)]` instead of `arr[seq]`. In the future this will be interpreted as an array index, `arr[np.array(seq)]`, which will result either in an error or a different result.
  vals = [arr_bpcs_complexity(arr[dims]) for dims in get_next_grid_dims(arr, grid_size)]
Grid 10000 of 34656
Grid 20000 of 34656
Grid 30000 of 34656
Traceback (most recent call last):
  File "C:\Users\CHE-Main\AppData\Local\Programs\Python\Python38-32\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\CHE-Main\AppData\Local\Programs\Python\Python38-32\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "D:\Documents\GitHub\bpcs\bpcs\bpcs.py", line 62, in <module>
    capacity(opts.infile, opts.outfile, opts.alpha)
  File "D:\Documents\GitHub\bpcs\bpcs\bpcs_steg_capacity.py", line 96, in capacity
    histogram(infile, outfile, alpha, greater)
  File "D:\Documents\GitHub\bpcs\bpcs\bpcs_steg_capacity.py", line 76, in histogram
    hist = x.modify(alpha, comp_fcn)
  File "D:\Documents\GitHub\bpcs\bpcs\bpcs_steg_capacity.py", line 53, in modify
    hist, navail, ntotal = histogram_of_complexity(self.arr, grid_size, alpha, comp_fcn)
  File "D:\Documents\GitHub\bpcs\bpcs\bpcs_steg_capacity.py", line 21, in histogram_of_complexity
    navail = sum([n for n, bin in zip(ns, bins) if comp_fcn(bin, alpha)])
  File "D:\Documents\GitHub\bpcs\bpcs\bpcs_steg_capacity.py", line 21, in <listcomp>
    navail = sum([n for n, bin in zip(ns, bins) if comp_fcn(bin, alpha)])
  File "D:\Documents\GitHub\bpcs\bpcs\bpcs_steg_capacity.py", line 95, in <lambda>
    greater = lambda x,thresh: x>=thresh
TypeError: '>=' not supported between instances of 'float' and 'NoneType'
```

### My Environment
Windows 10 2004 and Windows 10 1909
Python 3.8.3

### Side note
This library also seem to need `matplotlib` but it is not stated in the `requirements.txt`

**Comment by mobeets** [comment]
Ah, good find. I just pushed a change that should fix this issue. Could you run and confirm?

**Comment by Cerlancism** [comment]
Thank you. Looks like it's fixed . 

But I am not sure what is the range of the acceptable alpha values. I believe 0.01 ~ 0.99 should work. But if i put anything above 0.7 it now throws Assertion Error.

```
python -m bpcs.bpcs capacity -i examples/vessel.png -a 0.7
Slicing...
Graying...
Loaded image as array with shape (300, 300, 3, 8)
Creating histograms of image complexity...
Found 34656 grids
D:\Documents\GitHub\bpcs\bpcs\bpcs_steg_capacity.py:15: FutureWarning: Using a non-tuple sequence for multidimensional indexing is deprecated; use `arr[tuple(seq)]` instead of `arr[seq]`. In the future this will be interpreted as an array index, `arr[np.array(seq)]`, which will result either in an error or a different result.
  vals = [arr_bpcs_complexity(arr[dims]) for dims in get_next_grid_dims(arr, grid_size)]
Grid 10000 of 34656
Grid 20000 of 34656
Grid 30000 of 34656
3.0 of 34656.0 grids available with alpha of 0.7
Traceback (most recent call last):
  File "C:\Users\CHE-Main\AppData\Local\Programs\Python\Python38-32\lib\runpy.py", line 194, in _run_module_as_main
  File "C:\Users\CHE-Main\AppData\Local\Programs\Python\Python38-32\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "D:\Documents\GitHub\bpcs\bpcs\bpcs.py", line 62, in <module>
    capacity(opts.infile, alpha=opts.alpha, outfile=opts.outfile)
  File "D:\Documents\GitHub\bpcs\bpcs\bpcs_steg_capacity.py", line 96, in capacity
    histogram(infile, outfile, alpha, greater)
  File "D:\Documents\GitHub\bpcs\bpcs\bpcs_steg_capacity.py", line 76, in histogram
    hist = x.modify(alpha, comp_fcn)
  File "D:\Documents\GitHub\bpcs\bpcs\bpcs_steg_capacity.py", line 56, in modify
    nbytes = (get_n_message_grids([nbits_per_grid]*int(navail), int(navail))*nbits_per_grid)/8.0
  File "D:\Documents\GitHub\bpcs\bpcs\array_message.py", line 154, in get_n_message_grids
    assert x > 0
AssertionError
```

**Comment by mobeets** [comment]
Great, glad it works. I'll close this issue.

The range depends on the image. (It depends on how many blocks in the image have a "complexity" above the threshold you give.) So if you set it too high (0.7 in this case), there won't be enough complex blocks available to store any messages in. So the error you're seeing is, I'm fairly sure, an edge case I'm not handling for when this happens. In general, the lower the alpha, the larger a message you can store, but the more distorted the final image might look.
